### DISCUSSION TITLE
Keymap

### DISCUSSION THREAD

**Discussion by di-effe** [leftwm/leftwm#609] [Q&A]
Hello again.
Is there a smart way to create keymaps (for keybindings of course) and make them easaly available while working?

Not sure if the question is clear or make any sense, but being new with tiling I have to check frequently the documentation and I guess this is true for any new user.
I was thinking about parsing the config toml and use the result with something like rofi, but maybe it is work already done or an issue already solved in other ways.

Thanks

**Comment by mautamu** [comment]
You might be able to parse the output of `leftwm-check -v` if all you're looking for is a list of keybindings. As for creating them, it's been a longstanding issue (tagging #292), and I believe is being worked on in lefthk (https://github.com/VuiMuich/lefthk). (Tagging @VuiMuich).

Cheers,
Mautamu

**Reply by VuiMuich** [reply]
Well actually lefthk was initially created by Aethan, so far I just contributed some minor enhancements and did some beta testing.
Eventually I might get to implement an interactive keybind tui, but at this point no actual work has been done in this direction. So if anyone wants to get I to this, feel free!

**Reply by di-effe** [reply]
Using leftwm-check -v and a little of scripting I was able to reach this first step.
I am sure it can be done better and I´d like to group the bindings, but it's a start.

Also I am not 100% sure that rofi is the best choice to show this information.
Suggestions?
![Screenshot_12](https://user-images.githubusercontent.com/72241321/145006926-c094e14c-f09f-46f7-abe8-80aee2b6cb36.png)

**Reply by VuiMuich** [reply]
Nice!
I think grouping would be best done in the `config.toml` itself, right?
I how `leftwm check -v` treats comments in the config, but that might help as well to visualy improve this further.

**Reply by di-effe** [reply]
Or with some more code checking the command types and aggregating them in something that make sense.
Don´t know, I've to think more about it :)
